src/main/java/org/gridsuite/study/server/service/ConsumerService.java

```diff
@@ -398,367 +238,287 @@ public Consumer<Message<String>> consumeCaseImportFailed() {
+                        updateType = NotificationService.UPDATE_TYPE_DYNAMIC_SIMULATION_FAILED;
+                    } break;
                 }
+
+                UUID studyUuid = networkModificationTreeService.getStudyUuidForNodeId(receiverObj.getNodeUuid());
+                // send notification for failed computation
+                notificationService.emitStudyError(
+                        studyUuid,
+                        receiverObj.getNodeUuid(),
+                        updateType,
+                        errorMessage,
+                        userId);
+            } catch (JsonProcessingException e) {
+                LOGGER.error(e.toString());
             }
-        };
+        }
     }
 
-    @Bean
-    public Consumer<Message<String>> consumeSensitivityAnalysisStopped() {
-        return message -> {
-            String receiver = message.getHeaders().get(HEADER_RECEIVER, String.class);
-            if (receiver != null) {
-                NodeReceiver receiverObj;
-                try {
-                    receiverObj = objectMapper.readValue(URLDecoder.decode(receiver, StandardCharsets.UTF_8), NodeReceiver.class);
-
-                    LOGGER.info("Sensitivity analysis stopped for node '{}'", receiverObj.getNodeUuid());
+    public void consumeCalculationStopped(Message<String> msg, ComputationType computationType) {
+        String receiver = msg.getHeaders().get(HEADER_RECEIVER, String.class);
+        if (receiver != null) {
+            NodeReceiver receiverObj;
+            try {
+                receiverObj = objectMapper.readValue(URLDecoder.decode(receiver, StandardCharsets.UTF_8), NodeReceiver.class);
+
+                LOGGER.info("{} stopped for node '{}'", computationType.getLabel(), receiverObj.getNodeUuid());
+
+                String updateType = "";
+                // delete computation results from the database
+                switch (computationType) {
+                    case LOAD_FLOW: {
+                        networkModificationTreeService.updateLoadFlowResultUuid(receiverObj.getNodeUuid(), null);
+                        updateType = NotificationService.UPDATE_TYPE_LOADFLOW_STATUS;
+                    } break;
+                    case SECURITY_ANALYSIS: {
+                        networkModificationTreeService.updateSecurityAnalysisResultUuid(receiverObj.getNodeUuid(), null);
+                        updateType = NotificationService.UPDATE_TYPE_SECURITY_ANALYSIS_STATUS;
+                    } break;
+                    case SENSITIVITY_ANALYSIS: {
+                        networkModificationTreeService.updateSensitivityAnalysisResultUuid(receiverObj.getNodeUuid(), null);
+                        updateType = NotificationService.UPDATE_TYPE_SENSITIVITY_ANALYSIS_STATUS;
+                    } break;
+                    case SHORT_CIRCUIT: {
+                        // TODO : this Stopped function is the only one when SHORT_CIRCUIT doesn't handle buses => might cause a bug
```

Comment:
this todo is something you have discovred ?

Reply:
Yes I find it very suspicious but I am supposed to only refactor so I added a TODO.

Reply:
It's not possibe to stop a ONE_BUS shortcircuit for now, because it's not done with the button like other calculation. By the way, here in the server java code, nothing disallow to manage it properly now if we have HEADER_BUS_ID in the Header... but do we have it ?

Reply:
Nope you are right : HEADER_BUS_ID is only sent in case of success or failure. I remove the TODO.